presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -348,16 +348,21 @@ void PrestoServer::initializeAsyncCache() {
         kNumSsdShards,
         cacheExecutor_.get());
   }
-  auto memoryBytes = memoryGb << 30;
-
-  memory::MmapAllocator::Options options;
-  options.capacity = memoryBytes;
-  options.useMmapArena = systemConfig->useMmapArena();
-  options.mmapArenaCapacityRatio = systemConfig->mmapArenaCapacityRatio();
-
-  auto allocator = std::make_shared<memory::MmapAllocator>(options);
+  const auto memoryBytes = memoryGb << 30;
+
+  std::shared_ptr<memory::MemoryAllocator> baseAllocator;
+  if (systemConfig->useMmapAllocator()) {
+    memory::MmapAllocator::Options options;
+    options.capacity = memoryBytes;
+    options.useMmapArena = systemConfig->useMmapArena();
+    options.mmapArenaCapacityRatio = systemConfig->mmapArenaCapacityRatio();
+
+    baseAllocator = std::make_shared<memory::MmapAllocator>(options);
+  } else {
+    baseAllocator = memory::MemoryAllocator::createDefaultInstance();
+  }
   allocator_ = std::make_shared<cache::AsyncDataCache>(
```

Comment:
NYC: could you rename this to cache_? As it is mostly used as a cache in presto cpp. Then we can rename baseAllocator to allocaor